packages/plugins/graphql/server/services/content-api/index.js

```diff
@@ -70,6 +76,12 @@ module.exports = ({ strapi }) => {
 
       // Apply user-defined plugins
       plugins: extension.plugins,
+
+      // Generate artifacts only if one of the configuration has been set
+      shouldGenerateArtifacts: Object.values(outputs).some(negate(eq(false))),
```

Comment:
Also possible like this: 
```js
shouldGenerateArtifacts: Object.values(outputs).some(identity), 
```

Reply:
If you really want to look for defined why not use negate(isUndefined) or negate(isNill) ?

Reply:
The values cannot be nil since I've added a default `false` value when fetching them from the config.
That's why I'm only checking if they're not `false`. I can shorten it using `identity` or make it more transparent like `value => value !== false` if you want but other thank that, I'm not sure what else I could do :shrug:

Reply:
I honestly think we should not do this at all. But use the  process.env.NODE_ENV or allow setting it manually instead of using what's in outputs.

Reply:
Why not indeed, I've changed the behavior so that a user is able to set the value through the config + default value to true when in dev mode and false otherwise